extensions/bitwarden/src/search.tsx

```diff
@@ -0,0 +1,191 @@
+}
+
+function ItemList(props: { bitwardenApi: BitwardenApi; sessionToken: string | undefined }) {
+  const { bitwardenApi, sessionToken } = props;
+  const [state, setState] = useState<{ folders: Folder[]; items: Item[] }>();
+
+  const folderMap: Record<string, Folder> = {};
+  for (const folder of state?.folders || []) {
+    if (folder.id) folderMap[folder.id] = folder;
+  }
+
+  async function loadItems(sessionToken: string) {
+    try {
+      const [items, folders] = await Promise.all([
+        bitwardenApi.listItems<Item>("items", sessionToken),
+        bitwardenApi.listItems<Folder>("folders", sessionToken),
+      ]);
+
+      setState({ items, folders });
+    } catch (error) {
+      showToast(ToastStyle.Failure, "Failed to search vault", "Vault is locked");
+    }
+  }
+
+  useEffect(() => {
+    if (sessionToken) loadItems(sessionToken);
+  }, [sessionToken]);
+
+  return (
+    <List isLoading={typeof state === "undefined"}>
+      {state?.items.map((item) => (
+        <ItemListItem
+          key={item.id}
+          item={item}
+          folder={item.folderId ? folderMap[item.folderId] : undefined}
+          refreshItems={async () => {
+            if (sessionToken) {
+              const toast = await showToast(ToastStyle.Animated, "Syncing Items...");
+              await bitwardenApi.syncItems(sessionToken)
+              await loadItems(sessionToken);
+              await toast.hide();
+            }
+          }}
+        />
+      ))}
+    </List>
+  );
+}
+
+function ItemListItem(props: { item: Item; folder: Folder | undefined; refreshItems?: () => void }) {
+  const { item, folder, refreshItems } = props;
+  const { type, name, notes, identity, login, secureNote, fields, passwordHistory, card } = item;
+  const accessoryIcons = [];
+  if (folder) accessoryIcons.push(`📂 ${folder.name}`);
+  if (item.favorite) accessoryIcons.push(`⭐️`);
+
+  const fieldMap = Object.fromEntries(fields?.map((field) => [field.name, field.value]) || []);
+  const uriMap = Object.fromEntries(
+    login?.uris?.filter((uri) => uri.uri).map((uri, index) => [`uri${index + 1}`, uri.uri]) || []
+  );
+
+  let icon: string | Icon | undefined;
+  if (fetchFavicons && login?.uris?.[0]?.uri?.startsWith("https"))
+    icon = `https://s2.googleusercontent.com/s2/favicons?domain_url=${login?.uris?.[0].uri}`;
+  else
+    icon = {
+      1: Icon.Globe,
+      2: Icon.TextDocument,
+      3: Icon.List,
+      4: Icon.Person,
+    }[type];
+
+  const cleanItem = filterNullishPropertiesFromObject({
+    name,
+    notes,
+    identity: filterNullishPropertiesFromObject(identity),
+    login: filterNullishPropertiesFromObject(login),
+    card: filterNullishPropertiesFromObject(card),
+    secureNote,
+    fields,
+    folder: folder?.name,
+    passwordHistory,
+  });
+
+  const tree = treeify.asTree(cleanItem, true, false);
+
+  return (
+    <List.Item
+      id={item.id}
+      title={item.name}
+      keywords={item.name.split(".")}
```

Comment:
Can you think of any other characters we might want to `.split()` on?

Reply:
Maybe any non word characters ? I'm not sur if split works on regexes.

Reply:
Looks like [it does](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split)!